src/dg_triggers.cpp

```diff
@@ -976,7 +976,11 @@ int look_otrigger(ObjData *obj, CharData *actor, char *name, const char *additio
 
                 /* Don't allow a look to take place, if the object is purged. */
                 ret_val = (script_driver(&obj, t, OBJ_TRIGGER, TRIG_NEW) && obj);
-            }          
+                if (!ret_val)
+                  ret_val_0 = 1;
```

Comment:
You could replace this with ```return ret_val_0 && ret_val```

Reply:
Actually, it's slightly different... but what is the difference here?  Why not just 
``` ret_val = !ret_val``` if you're just going to toggle them.

Reply:
I'm not toggling them.  Currently, the ret_val gets set by every trigger that runs, then only returns the value from the very last trigger.  So that means if a trigger early in the cascade returns 0 (which should block the object description from being printed to the character), if any trigger after that returns another value, ret_val gets set to that later value, over-writing the return 0 from the earlier trigger and printing the object description even if it wasn't intended to.

Because I do want every trigger that might be applicable to the targeted object and argument to run, I don't want to have the function return 0 and bail if a trigger returns 0.  But I also can't just put any return 0 function at the end of the object's trigger assignments.

So what I've done is created a variable that gets set if a trigger returns 0 at any point, and then sets the ret_val back to 0 after every trigger after it runs, ensuring that the final ret_val is 0.

Does that make sense?  Is there a better way to do this?

Reply:
0 is a fail - do not print the object description.

1 is a success - do print the object description.

I want to return 0 if ANY trigger returns 0.

There are no options other than 0 or 1, it is either fail or succeed.

Reply:
I've updated the code to match Strider's suggestion.  Works exactly as intended now.